Content.Server/GameObjects/EntitySystems/EmitSoundSystem.cs

```diff
@@ -0,0 +1,99 @@
+using Content.Server.GameObjects.Components.Sound;
+using Content.Shared.Audio;
+using Content.Shared.Interfaces.GameObjects.Components;
+using JetBrains.Annotations;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    [UsedImplicitly]
+    public class EmitSoundSystem : EntitySystem
+    {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+
+        /// <inheritdoc />
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<EmitSoundOnLandComponent, LandEvent>((eUI, comp, arg) => PlaySoundBasedOnMode(eUI, comp));
+            SubscribeLocalEvent<EmitSoundOnUseComponent, UseInHandEvent>((eUI, comp, arg) => PlaySoundBasedOnMode(eUI, comp));
+            SubscribeLocalEvent<EmitSoundOnThrowComponent, ThrownEvent>((eUI, comp, arg) => PlaySoundBasedOnMode(eUI, comp));
+            SubscribeLocalEvent<EmitSoundOnActivateComponent, ActivateInWorldEvent>((eUI, comp, args) => PlaySoundBasedOnMode(eUI, comp));
+        }
+
+        private void PlaySoundBasedOnMode(EntityUid uid, BaseEmitSoundComponent component)
+        {
+            if (component == null)
```

Comment:
pretty sure component can't be null here since its a directed event, but keep it if its yelling at you for null references